workitem/workitem_repository.go

```diff
@@ -136,40 +135,29 @@ func (r *GormWorkItemRepository) CalculateOrder(above, below *float64) float64 {
 //      FindSecondItem returns the value above which reorder item has to be placed
 func (r *GormWorkItemRepository) FindSecondItem(order *float64, secondItem string, WorkItemId string) (*string, *float64, error) {
 	Item := WorkItem{}
+	var tx *gorm.DB
 	switch secondItem {
 	case below:
 		// Finds the item below which reorder item has to be placed
-		tx := r.db.Where("executionorder < ?", order).Order("executionorder desc", true).Last(&Item)
-		if Item.ID == 0 {
-			// Item is placed at first position
-			ItemId := strconv.FormatUint(Item.ID, 10)
-			return &ItemId, &Item.Executionorder, nil
-		} else if tx.Error != nil {
-			return nil, nil, tx.Error
-		} else if tx.RecordNotFound() {
-			return nil, nil, tx.Error
-		} else {
-			ItemId := strconv.FormatUint(Item.ID, 10)
-			return &ItemId, &Item.Executionorder, nil
-		}
+		tx = r.db.Where("executionorder < ?", order).Order("executionorder desc", true).Last(&Item)
 	case above:
 		// Finds the item above which reorder item has to be placed
-		tx := r.db.Where("executionorder > ?", order).Order("executionorder", true).Last(&Item)
-		if Item.ID == 0 {
-			// Item is placed at last position
-			ItemId := strconv.FormatUint(Item.ID, 10)
-			return &ItemId, &Item.Executionorder, nil
-		} else if tx.Error != nil {
-			return nil, nil, tx.Error
-		} else if tx.RecordNotFound() {
-			return nil, nil, tx.Error
-		} else {
-			ItemId := strconv.FormatUint(Item.ID, 10)
-			return &ItemId, &Item.Executionorder, nil
-		}
+		tx = r.db.Where("executionorder > ?", order).Order("executionorder", true).Last(&Item)
 	default:
 		return nil, nil, nil
 	}
+	if Item.ID == 0 {
+		// Item is placed at first position
+		ItemId := strconv.FormatUint(Item.ID, 10)
+		return &ItemId, &Item.Executionorder, nil
+	} else if tx.Error != nil {
+		return nil, nil, tx.Error
+	} else if tx.RecordNotFound() {
+		return nil, nil, tx.Error
+	} else {
+		ItemId := strconv.FormatUint(Item.ID, 10)
+		return &ItemId, &Item.Executionorder, nil
+	}
```

Comment:
There's no return after this last `else` block so you can just omit the `else` all together and just return without further nesting.